internal/fsutil/create.go

```diff
@@ -48,8 +48,15 @@ func Create(options *CreateOptions) (*Entry, error) {
 
 	switch o.Mode & fs.ModeType {
 	case 0:
-		err = createFile(o)
-		hash = hex.EncodeToString(rp.h.Sum(nil))
+		if o.Link != "" {
+			// Creating the hard link does not occurs file reads. Therefore,
+			// its size and hash is not calculated here but in
+			// `testutil.TreeDumpEntry` for testing purpose instead.
```

Comment:
This is not valid per what we discussed. If it was, I don't think this is the best place for the comment, it is talking about a completely different function which is only used for testing.

Reply:
I reckon this is the case --- Creating the hard link using the syscall is only updating the metadata of a file, which does not read the file body.

Reply:
Anyhow, please do not resolve the conversations if there is still discussion going on. In this case, this is not the place for the comment. Think about it, you are saying to look at non-exported test function in another package and you are describing the usage of the tests, if you wanted to get the size and hash from another module you would not care about `DumpEntry`.

Please remove it from here.

Reply:
I think we need to keep 
```
			// Creating the hard link does not occurs file reads.
			// Therefore, its size and hash is not calculated here.
```
here, since the size and the hash will also be used in the future by chisel db (correct me if I misunderstood that).

Reply:
Probably this should be documented on the public interface but let's keep it like this for now.